app/controllers/recipe_foods_controller.rb

```diff
@@ -1,54 +1,56 @@
 class RecipeFoodsController < ApplicationController
-  before_action :set_recipe
+  before_action :set_recipe_food, only: %i[show edit destroy]
 
-  def new
-    @recipe_food = @recipe.recipe_foods.new
+  # GET /recipe_foods or /recipe_foods.json
+  def index
+    @recipe_foods = RecipeFood.all
   end
 
-  def create
-    @recipe_food = @recipe.recipe_foods.find_or_initialize_by(food_id: recipe_food_params[:food_id])
-    @recipe_food.quantity = recipe_food_params[:quantity] # Set quantity
-    @recipe_food.calculate_value
-    if @recipe_food.save
-      redirect_to @recipe, notice: 'Food added successfully.'
-    else
-      render :new
-    end
+  # GET /recipe_foods/1 or /recipe_foods/1.json
+  def show; end
```

Comment:
Hello, this is Temitope. I'm here to help you with your code. 😊

**NameError in Recipes#show**

- It appears there is a `NameError` due to an undefined local variable or method `general_shopping_lists_path`. Ensure that you have defined this path in your routes file, and that it is spelled correctly both there and in your view file. Also, make sure any necessary controllers or methods needed for this path are properly defined and accessible.
- To avoid this error in the future, you can use the `rake routes` command to check the available paths in your application, and use the `*_path` or `*_url` helpers to generate the correct links in your views. You can also use the `link_to` helper with a block to create more complex links with HTML elements.
- For more information on how to use routing and links in Rails, you can check out some of the web search results I found for you   , or ask me for more tips on how to debug or improve your code. 😊
<img width="837" alt="image" src="https://github.com/chimwemwe007/my-recipe-app/assets/52934128/a9844aa5-5fd4-4181-99f7-205aeb8e69c4">